ext/node/ops/vm_internal.rs

```diff
@@ -0,0 +1,595 @@
+  }
+
+  pub fn from_sandbox_obj<'a>(
+    scope: &mut v8::HandleScope,
+    sandbox_obj: v8::Local<v8::Object>,
+  ) -> Option<&'a Self> {
+    let private_str =
+      v8::String::new_from_onebyte_const(scope, &PRIVATE_SYMBOL_NAME);
+    let private_symbol = v8::Private::for_api(scope, private_str);
+
+    sandbox_obj
+      .get_private(scope, private_symbol)
+      .and_then(|wrapper| {
+        deno_core::cppgc::try_unwrap_cppgc_object::<Self>(wrapper)
+      })
+  }
+
+  pub fn is_contextify_context(
+    scope: &mut v8::HandleScope,
+    object: v8::Local<v8::Object>,
+  ) -> bool {
+    Self::from_sandbox_obj(scope, object).is_some()
+  }
+
+  pub fn context<'a>(
+    &self,
+    scope: &mut v8::HandleScope<'a>,
+  ) -> v8::Local<'a, v8::Context> {
+    v8::Local::new(scope, &self.context)
+  }
+
+  fn global_proxy<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> v8::Local<'s, v8::Object> {
+    let ctx = self.context(scope);
+    ctx.global(scope)
+  }
+
+  fn sandbox<'a>(
+    &self,
+    scope: &mut v8::HandleScope<'a>,
+  ) -> v8::Local<'a, v8::Object> {
+    v8::Local::new(scope, &self.sandbox)
+  }
+
+  fn get<'a, 'c>(
+    scope: &mut v8::HandleScope<'a>,
+    object: v8::Local<'a, v8::Object>,
+  ) -> Option<&'c ContextifyContext> {
+    let Some(context) = object.get_creation_context(scope) else {
+      return None;
+    };
+
+    let context_ptr = context.get_aligned_pointer_from_embedder_data(0);
+    // SAFETY: We are storing a pointer to the ContextifyContext
+    // in the embedder data of the v8::Context during creation.
+    Some(unsafe { &*(context_ptr as *const ContextifyContext) })
+  }
+}
+
+pub const VM_CONTEXT_INDEX: usize = 0;
+
+fn create_v8_context<'a>(
+  scope: &mut v8::HandleScope<'a>,
+  object_template: v8::Local<v8::ObjectTemplate>,
+  snapshot_data: Option<&'static [u8]>,
+) -> v8::Local<'a, v8::Context> {
+  let scope = &mut v8::EscapableHandleScope::new(scope);
+
+  let context = if let Some(_snapshot_data) = snapshot_data {
+    v8::Context::from_snapshot(scope, VM_CONTEXT_INDEX).unwrap()
+  } else {
+    v8::Context::new_from_template(scope, object_template)
+  };
+
+  scope.escape(context)
+}
+
+#[derive(Debug, Clone)]
+struct SlotContextifyGlobalTemplate(v8::Global<v8::ObjectTemplate>);
+
+fn init_global_template<'a>(
+  scope: &mut v8::HandleScope<'a>,
+) -> v8::Local<'a, v8::ObjectTemplate> {
+  let mut maybe_object_template_slot =
+    scope.get_slot::<SlotContextifyGlobalTemplate>();
+
+  if maybe_object_template_slot.is_none() {
+    init_global_template_inner(scope);
+    maybe_object_template_slot =
+      scope.get_slot::<SlotContextifyGlobalTemplate>();
+  }
+  let object_template_slot = maybe_object_template_slot
+    .expect("ContextifyGlobalTemplate slot should be already populated.")
+    .clone();
+  v8::Local::new(scope, object_template_slot.0)
+}
+
+extern "C" fn c_noop(_: *const v8::FunctionCallbackInfo) {}
+
+thread_local! {
```

Comment:
Can you link to the note from https://github.com/denoland/deno/blob/9b34b7ed1862c4b7bd84d52cd3417f5a1592efc4/ext/node/global.rs#L12?